consumer.go

```diff
@@ -214,6 +207,14 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 
 			shardIterator = resp.NextShardIterator
 		}
+
+		// Wait for next scan
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-scanTicker.C:
```

Comment:
ohh, do we need to reset the ticker?

Reply:
Not sure if there's a way to reset a ticker without recreating it?  In any case, this is the current behavior AFAICT: the duration of the Kinesis API call eats into the scan interval.